apple/internal/apple_framework_import.bzl

```diff
@@ -63,68 +91,103 @@ def _apple_framework_import_impl(ctx):
-    )
+    return AppleFrameworkImportInfo(**provider_fields)
+
+def _apple_dynamic_framework_import_impl(ctx):
+    """Implementation for the apple_dynamic_framework_import rule."""
+    providers = []
+
+    transitive_sets = _transitive_framework_imports(ctx.attr.deps)
+    filtered_framework_imports = filter_framework_imports_for_bundling(ctx.files.framework_imports)
+    if filtered_framework_imports:
+        transitive_sets.append(depset(filtered_framework_imports))
+    providers.append(_framework_import_info(transitive_sets))
+
+    framework_groups = _grouped_framework_files(ctx.files.framework_imports)
+    framework_dirs_set = depset(framework_groups.keys())
+    objc_provider = _objc_provider_with_dependencies(ctx, {
+        "dynamic_framework_file": depset(ctx.files.framework_imports),
+        "dynamic_framework_dir": framework_dirs_set,
+    })
+    providers.append(objc_provider)
 
     # TODO(kaipi): Remove this dummy binary. It is only required because the
     # new_dynamic_framework_provider Skylark API does not accept None as an argument for the binary
     # argument. This change was submitted in https://github.com/bazelbuild/bazel/commit/f8ffac. We
     # can't remove this until that change is released in bazel.
     dummy_binary = ctx.actions.declare_file("_{}.dummy_binary".format(ctx.label.name))
     ctx.actions.write(dummy_binary, "_dummy_file_")
+    providers.append(apple_common.new_dynamic_framework_provider(
+        binary = dummy_binary,
+        objc = objc_provider,
+        framework_dirs = framework_dirs_set,
+        framework_files = depset(ctx.files.framework_imports),
+    ))
+
+    return providers
+
+def _apple_static_framework_import_impl(ctx):
+    """Implementation for the apple_static_framework_import rule."""
+    providers = []
+
+    transitive_sets = _transitive_framework_imports(ctx.attr.deps)
+    providers.append(_framework_import_info(transitive_sets))
+
+    framework_imports = ctx.files.framework_imports
+    objc_provider_fields = {
+        "static_framework_file": depset(framework_imports),
+    }
+
+    framework_groups = _grouped_framework_files(ctx.files.framework_imports)
+    if ctx.attr.alwayslink:
+        objc_provider_fields["force_load_library"] = depset(
+            _all_framework_binaries(framework_groups),
+        )
+    if ctx.attr.sdk_dylibs:
+        objc_provider_fields["sdk_dylib"] = depset(ctx.attr.sdk_dylibs)
+    if ctx.attr.sdk_frameworks:
+        objc_provider_fields["sdk_framework"] = depset(ctx.attr.sdk_frameworks)
+    if ctx.attr.weak_sdk_frameworks:
+        objc_provider_fields["weak_sdk_framework"] = depset(ctx.attr.weak_sdk_frameworks)
+
+    providers.append(_objc_provider_with_dependencies(ctx, objc_provider_fields))
+
+    bundle_files = [x for x in framework_imports if ".bundle/" in x.short_path]
+    if bundle_files:
+        parent_dir_param = partial.make(
+            resources.bundle_relative_parent_dir,
```

Comment:
CI is failing because `resources` is now undefined

Reply:
yeah, I meant to ask to remove the `AppleResourceBundleInfo` import above, not the `resources` one, my mistake

Reply:
I had mistakenly looked for `resources(` instead of `resources.`